### DISCUSSION TITLE
Jellyfin links in GoogleTV search results?

### DISCUSSION THREAD

**Discussion by zehnerGIT** [jellyfin/jellyfin-androidtv#4120] [General]
I have one of those SEI804 clones, the Nokia 8010. With Android TV 11 and Jellyfin 0.16 the google assistant listed jellyfin direct links in the results page (e.g. if I say "Tron film" I got a page with synopsis, cast&crew, rotten tomatoes rating, trailers ... and a link to the jellyfin details page if the movie is part of my library). But after the update to Android 12 (or Jellyfin 0.17, both updates were very close, so I don't know, when the behaviour changed), the Jellyfin search result is missing.

Do someone here still have the link in the GoogleTV overview page (in Android 12 or later) or is this a general problem with the Android or Jellyfin version?

The voice search inside the app still works, and jellyfin is enabled as app for search. I've reinstalled the app, cleared data and cache, but to no avail.

**Comment by nielsvanvelzen** [comment]
On newer versions of the Android TV platform the "launcher" app (what the home button brings you to) changed to Google TV. When using Google TV the search uses a cloud-based solution from Google that Jellyfin can't hook into because it would require a centralized server (which we're strongly opposed to!).

So unfortunately that means no search from the launcher anymore.

**Reply by newton181** [reply]
For this reazon i'm stucked in Anndroid TV 9 and buy Android TV 12 and earlier tv boxes, Google TV is and ad-based launcher, google nerfed Android TV to $how us ad$.

> On newer versions of the Android TV platform the "launcher" app (what the home button brings you to) changed to Google TV. When using Google TV the search uses a cloud-based solution from Google that Jellyfin can't hook into because it would require a centralized server (which we're strongly opposed to!).
> 
> So unfortunately that means no search from the launcher anymore.

**Reply by zehnerGIT** [reply]
> On newer versions of the Android TV platform the "launcher" app (what the home button brings you to) changed to Google TV. When using Google TV the search uses a cloud-based solution from Google that Jellyfin can't hook into because it would require a centralized server (which we're strongly opposed to!).
> 
> So unfortunately that means no search from the launcher anymore.

Thanks for the reply. It's really a pity, that Android TV gets more ads and less features with every new version, but a few new Android 12 features are essential (e.g. automatic framerate matching) and work at least in the Jellyfin app, so I have to stay with Android 12

**Reply by Carve** [reply]
@nielsvanvelzen Are there any launchers we could potentially switch to that would fix or allow this?

**Reply by nielsvanvelzen** [reply]
Third party launchers do not have access to the search API without rooting (ask me [how I know](https://github.com/nielsvanvelzen/tv-launcher)). I'm not aware of any launcher implementing search.
I'd recommend using a very minimal launcher (or enable "app only mode" on Google TV) and search from within the Jellyfin app.

**Comment by phirestalker** [comment]
They want a file on the cloud to get the media metadata from, so why can't we just use a generic one from you guys? You could just have a handful of things on there to look legit. This way all of us self-hosters, can use DNS rules to point that URL to our local server, and viola! Am I missing something? Would this even be possible?
